contents/docs/libraries/vue-js.md

```diff
@@ -0,0 +1,281 @@
+```
+
+### Initializing PostHog
+
+We will cover three different methods for initializing PostHog:
+
+1. [Plugins](#method-1-create-a-plugin)
+2. [Provide/inject](#method-2-use-provide--inject)
+3. [Vue.prototype](#method-3-use-vueprototype)
+
+Choose what is best for you, considering your Vue version and your codebase’s stylistic choices.
+
+### Method 1: Create a Plugin
+
+_Note: For both Vue 3.x and Vue 2.x users_
+
+First, create a new file `posthog.js` in your plugins directory.
+
+```bash
+mkdir plugins #skip if you already have one
+cd plugins 
+touch posthog.js 
+```
+
+Next, create a plugin and assign PostHog to Vue’s global properties. The code will differ depending on your version of Vue. 
+
+**Vue 3.x:** 
+
+```js
+//./plugins/posthog.js
+import posthog from "posthog-js";
+
+export default {
+  install(app) {
+    app.config.globalProperties.$posthog = posthog.init(
+      "<ph_project_api_key>",
+      {
+        api_host: "<ph_instance_address>",
+      }
+    );
+  },
+};
+```
+
+**Vue 2.x:** 
+
+```js
+//./plugins/posthog.js
+import posthog from "posthog-js";
+
+export default {
+  install(Vue, options) {
+    Vue.prototype.$posthog = posthog.init(
+      "<ph_project_api_key>",
+      {
+        api_host: "<ph_instance_address>"
+      }
+    );
+  }
+};
+```
+
+Finally, activate your plugin in the file where you initialize your app. For Vue 3.x, it will likely be your `index.js` file. For Vue 2.x, it will be your `main.js` file.
+
+**Vue 3.x:** 
+
+```js
+//main.js
+import { createApp } from 'vue'
+import App from './App.vue'
+import posthogPlugin from "./plugins/posthog"; //import the plugin. 
+
+const app = createApp(App);
+
+app.use(posthogPlugin); //install the plugin
+app.mount('#app')
+```
+
+**Vue 2.x:** 
+
+```js
+//main.js
+import posthogPlugin from "./plugins/posthog"; // import the plugin
+
+Vue.use(posthogPlugin); // install the plugin, before new Vue()
+```
+
+You can now use PostHog throughout your Vue app using `this.$posthog`.  For example, 
+
+```js
+//component.vue
+
+<script>
+export default {
+  data() {
+    return {
+      foo: "bar!",
+    };
+  },
+  watch: {
+    // whenever question changes, this function will run
+    foo(newFoo, oldFoo) {
+      this.$posthog.capture("Foo Changed", {foo: newFoo});
+    },
+  },
+  created() {
+    console.log("Created", this.$posthog); //posthog accessible anywhere!
+  },
+};
+</script>
+```
+
+### Method 2: Use provide / inject
+
+_Note: For Vue 3.x users only_
+
+With Vue 3.x, developers can use `provide()` and `inject()` to pipe global values into any component without prop drilling. And if you don’t know what prop drilling is, good for you.
+
+While this method is more declarative — as you need to inject PostHog into every component — it avoids “polluting” globals (such as in Method 1). Some engineers prefer this approach, while others to include PostHog in globals since it doesn’t need to be reactive and will be called throughout your application. 
```

Comment:
```suggestion
While this method is more declarative — as you need to inject PostHog into every component — it avoids “polluting” globals (like method 1 does). Some engineers prefer this approach, while others include PostHog in globals since it doesn’t need to be reactive and will be called throughout your application. 
```